net/ieee802154: Split the net mgmt part from the L2 specific header

tbursztyka opened the pull request:
Based on findings from @SebastianBoe  ieee802154 is exposing to widely L2 private data.

Commit:
net/ieee802154: Split the net mgmt part from the L2 specific header

L2 specific data and IEEE 802154 net mgmt interface are not related.
Plus, application may use the net mgmt part, not the L2 one. So let's
split the content in relevant headers.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

dbkinder commented on include/net/ieee802154_mgmt.h:
```diff
@@ -0,0 +1,233 @@
+
+/**
+ * @brief IEEE 802.15.4 library
+ * @defgroup ieee802154 IEEE 802.15.4 Library
```
defgroup ``iee802154`` is also still defined in iee802154.h  I suspect you should delete that defgroup or change that one to an addtogroup to avoid confusion and a potential error.

dbkinder reviewed (changes requested):
duplicate @defgroup name

Commit:
net/ieee802154: Do not expose l2 specific context in generic net_l2.h

This will avoid exposing IEEE 802.15.4 Zephyr's L2 private context data
to unrelevant places.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

carlescufi commented:
@dbkinder we depend on this PR for the upcoming OpenThread one. Could you take a look again?

SebastianBoe reviewed (approved):
This resolves the dependency issues I was facing with OpenThread.
